### PR TITLE
Throw an error on workspace start if ssl_exception error received

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Throw an error on workspace start if ssl_exception error received

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-6940

### Is it tested? How?

see https://github.com/eclipse-che/che-server/pull/710

### CODE REVIEW

File: packages/dashboard-frontend/src/components/WorkspaceProgress/CreatingSteps/Initialize/index.tsx [hunk 1]
```diff
@@ -149,6 +149,8 @@ class CreatingStepInitialize extends ProgressStep<Props, State> {
       throw new Error(
         'Could not resolve devfile from private repository because authentication request is missing a parameter, contains an invalid parameter, includes a parameter more than once, or is otherwise invalid.',
       );
+    } else if (errorCode === 'ssl_exception') {
```
Comment (olexii4) [inline packages/dashboard-frontend/src/components/WorkspaceProgress/CreatingSteps/Initialize/index.tsx:152]: @vinokurig These changes required an additional test.

`packages/dashboard-frontend/src/components/WorkspaceProgress/CreatingSteps/Initialize/__tests__/index.spec.tsx`

```
...
test('`ssl_exception` error code', async () => {
    const searchParams = new URLSearchParams({
      [FACTORY_URL_ATTR]: factoryUrl,
      [ERROR_CODE_ATTR]: 'ssl_exception',
    });

    renderComponent(store, searchParams);

    await jest.advanceTimersByTimeAsync(MIN_STEP_DURATION_MS);

    const expectAlertItem = expect.objectContaining({
      title: 'Failed to create the workspace',
      children: expect.stringContaining(
        'SSL handshake failed. Please, contact the cluster administrator.',
      ),
      actionCallbacks: [
        expect.objectContaining({
          title: 'Click to try again',
          callback: expect.any(Function),
        }),
      ],
    });
    await waitFor(() => expect(mockOnError).toHaveBeenCalledWith(expectAlertItem));

    expect(mockOnNextStep).not.toHaveBeenCalled();
  });
  ...
```
Could you fix it?

Comment (vinokurig) [reply packages/dashboard-frontend/src/components/WorkspaceProgress/CreatingSteps/Initialize/index.tsx:152]: Done, thanks for the code snippet.

**Review by olexii4 — APPROVED** [review]
LGTM

**Comment by artaleks9** [comment]
Verified on Eclipse Che with `quay.io/eclipse/che-dashboard:pr-1171` and `quay.io/eclipse/che-server:pr-710`
Checked using gitlab server with SSl certificate https://gitlab-gitlab-system.apps.git.crw-qe.com

**The functionality works properly**
Eclipse-Che-Creating-a-workspace-pr-710

**Review by akurinnoy — APPROVED** [review]

**Pull request merged**